### DISCUSSION TITLE
Prevent X being passed to a forecaster

### DISCUSSION THREAD

**Discussion by hliebert** [alan-turing-institute/sktime#5768] [General]
Is there a mechanism that can be used to prevent `X` being passed to a model? 

Consider a case where both y and X are passed to cross-validation, which nests `TransformedTargetForecaster` and  `ForecastingPipeline`, and different forecasters are substituted as the final forecaster in the pipe during cross validation.

Some models take optional `X` features (e.g. `ARIMA`). However, I do not want to pass `X` to some of those models. Is there a simple way to prevent this?  

I've tried to add `FeatureSelection` to `ForecastingPipeline`, and adding `method="none"` as a parameter during CV.  However, this causes an Exception since the output of `transform` should be `pd.Series` or `pd.DataFrame` and cannot be `None`. I assume this is a bug, unless there is another use case for `FeatureSelection(method="none")` that I am missing.

`TypeError X: _transform output of <class 'sktime.transformations.series.feature_selection.FeatureSelection'> does not comply with sktime mtype specifications. See datatypes.MTYPE_REGISTER for mtype specifications. Returned error message: ["obj must be a pandas.DataFrame, found <class 'NoneType'>", "obj must be a pandas.Series, found <class 'NoneType'>", "obj must be a numpy.ndarray, found <class 'NoneType'>"]. Returned object: None`

**Comment by fkiraly** [comment]
Interesting!

There is a way I can think of: if you take your forecaster and do `forecaster.set_tags(**{"ignores-exogeneous-X": True})`, it will ignore `X`. However, this may be flaky, as some operations may reset dynamic tags, and it is non-idiomatic, as this choice is not exposed as a hyper-parameter.

I will write a forecaster for you that does this, and think more genereal about how best to let users specify tags.

> However, this causes an Exception since the output of transform should be `pd.Series` or `pd.DataFrame` and cannot be `None`. I assume this is a bug, unless there is another use case for `FeatureSelection(method="none")` that I am missing.

This is interesting - I think a `None` output should be allowed, but it is a bit of a corner case and I am not sure whether tests cover that really well. Reproducible code in an issue would be appreciated.

**Reply by fkiraly** [reply]
Here is a compositor, would be keen to hear what you think:

https://github.com/sktime/sktime/pull/5769

Using the compositor exposes the choice on whether to pass on `X` as a tunable parameter.

**Reply by hliebert** [reply]
Thanks for the quick PR! I'll test it and report back.

Here is a reproducible example for the problem with `FeatureSelection(method="none")`. 

```python
import numpy as np
from sktime.forecasting.arima import ARIMA
from sktime.datasets import load_longley
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.compose import ForecastingPipeline
from sktime.transformations.series.feature_selection import FeatureSelection

y, X = load_longley()
horizon = ForecastingHorizon(np.arange(1, 4), is_relative=True)

pipe = ForecastingPipeline(
    [
        ("select_X", FeatureSelection(method="none")),
        ("arima", ARIMA()),
    ]
)

pipe.fit(y, X, fh=horizon)
```

**Reply by fkiraly** [reply]
Thanks! It was a minimal change to allow `None` return, so I did that and added the example above as a test: https://github.com/sktime/sktime/pull/5772

**Comment by yarnabrina** [comment]
May I ask why do you not want to use `X` if the algorithm supports it? Usually that leads to better results, so curious what is your use case?

**Reply by hliebert** [reply]
The use case is twofold.

- We want to assess the performance of a simple time series model without additional features as a benchmark. 
- In the current case the feature set is moderately large and contains lags which are fed to a tabular reducer. Models like `ARIMA` do not cope well with this. I agree that having a set of smaller set of dedicated `X` for those models may be helpful, but we are not there yet.
